### PR TITLE
Support constructing DRep update certificates

### PR DESCRIPTION
**Pull request by carlhammann**
# Changelog

```yaml
- description: |
    Support constructing DRep update certificates
# uncomment types applicable to the change:
  type:
  - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

This goes towards solving [this issue](https://github.com/IntersectMBO/cardano-cli/issues/550) on the cardano-cli, which already has an open PR [here](https://github.com/IntersectMBO/cardano-cli/pull/561).

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]

File: cardano-api/internal/Cardano/Api/Certificate.hs [hunk 1]
```diff
@@ -465,6 +467,21 @@ makeStakeAddressAndDRepDelegationCertificate w cred delegatee deposit =
         delegatee
         (toShelleyLovelace deposit)
 
+data DRepUpdateRequirements era where
```
Comment (carbolymer) [inline cardano-api/internal/Cardano/Api/Certificate.hs:470]: This type feels a bit artificial - it's used only to call this function. Why not a separate witness and credential?

Comment (smelc) [reply cardano-api/internal/Cardano/Api/Certificate.hs:470]: @carbolymer> a few GADTs in the file have two constructors, distinguishing pre-conway and conway onwards. Other GADTs have only constructor and coud likely be bypassed, like the one @carlhammann introduced.

But for the sake of consistency, I would keep it like Carl wrote it. It's consistent with the file's existing style (that was mainly written by @Jimbo4350 and @newhoggy).

Comment (carbolymer) [reply cardano-api/internal/Cardano/Api/Certificate.hs:471]: @smelc let's leave it as it is then!

**Comment by smelc** [comment]
Took the only commit on my sole authorship, because I couldn't sign it for Carl when rebasing :upside_down_face:

**Pull request merged**